.circleci/config.yml

```diff
@@ -108,49 +142,25 @@ jobs:
     environment:
       PYTHON: 3.6.5
       HOMEBREW_NO_AUTO_UPDATE: 1
+      TEMP_FILE: temp_requirements_summary.txt
+      IGNORED_DIRS: envs|tests|circuit-fault-diagnosis|factoring
 
   test-osx-3.5:
     <<: *osx-tests-template
     environment:
       PYTHON: 3.5.5
       HOMEBREW_NO_AUTO_UPDATE: 1
+      TEMP_FILE: temp_requirements_summary.txt
```

Comment:
I tried to move the environment variables one level up (i.e. to the same level as the builds), but CircleCI just seem to think that `environment` was the name of a build. Based on their code examples for environment variables and on the experiment below, I don't think I can move the `environment` up one level.

Code in commit 445e861
```
jobs:
  environment:
    SUM_REQ_FILE: temp_requirements_summary.txt
    IGNORED_DIRS: envs|tests|circuit-fault-diagnosis|factoring

  test-3.7: &full-test-template
    docker:
..
```

Returned error
```
Build-agent version 1.0.10445-2137dfde (2019-04-10T16:41:04+0000)
Configuration errors: 1 error occurred:

* Error migrating config to version 2: 2 errors occurred:

* in job 'environment': steps is not a list
* in job 'environment': steps is not a list
```

Reply:
Can't you just make environment top-level, global?

Reply:
Actually, what you have already, per-job, works great.

Reply:
I was confused because you have variables repeated for every OS X job (wouldn't just one definition in the template be enough?)

Reply:
Yes, I had also initially thought that one definition in the macOS template would be enough (see commit 596076f).

However, since the macOS tests need to overwrite the template's environment.PYTHON variable, the entire `environment` step gets overwritten. (i.e. `SUM_REQ_FILE` and `IGNORED_DIRS` do not exist for the non-template macOS tests).

Hence, I had repeated the `SUM_REQ_FILE` and `IGNORED_DIRS` on each use of the macOS `environment`. (commit cf22291)

Ideally, I would prefer if `HOMEBREW_NO_AUTO_UPDATE`, `SUM_REQ_FILE`, and `IGNORED_DIRS` were only defined in the template, and only `PYTHON` was edited on each macOS test.

Reply:
I agree, that would be the best.

Could "reusable executors" [help](https://circleci.com/docs/2.0/reusing-config/#overriding-keys-when-invoking-an-executor) here?

Reply:
Yes, "reusable executors" will help here. I read that it's additive, so it won't overwrite all environment variables (like the issue outlined in my previous comment).

I would prefer to implement reusable executors at a later time as it seems like a major refactor. Namely, upgrading to CircleCI version 2.1 (which has reusable executors), and replacing our use of templates to reusable executors. In the meantime, I have made a GitHub issue for this, #50

Reply:
Sounds good. Thanks!